examples/fodo_channel_restart/run_fodo.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022-2024 ImpactX contributors
+# Authors: Axel Huebl, Chad Mitchell, Marco Garten
+# License: BSD-3-Clause-LBNL
+#
+# -*- coding: utf-8 -*-
+
+from impactx import ImpactX, elements, push
+
+sim = ImpactX()
+
+# set numerical parameters and IO control
+sim.particle_shape = 2  # B-spline order
+sim.space_charge = False
+# sim.diagnostics = False  # benchmarking
+sim.slice_step_diagnostics = True
+
+# domain decomposition & space charge mesh
+sim.init_grids()
+
+# load a 2 GeV electron beam with an initial
+# unnormalized rms emittance of 2 nm
+kin_energy_MeV = 2.0e3  # reference energy
+bunch_charge_C = 1.0e-9  # used with space charge
+npart = 10000  # number of macro particles
+
+#   reference particle
+pc = sim.particle_container()
+ref = pc.ref_particle()
+ref.set_charge_qe(-1.0).set_mass_MeV(0.510998950).set_kin_energy_MeV(kin_energy_MeV)
+
+#   load particle bunch from file
+push(pc, elements.Source("../FODO_channel/diags/openPMD/monitor.h5"))
```

Comment:
Depending on the use case, e.g., checkpoint-restart, it can be useful here to also (re-)set the `s` position of the reference particle -- and the whole reference particle per se.

We could add a runtime option for this.

Reply:
By this, I think you mean--to read in the reference particle data (at the specified s location), and to use this to initialize the reference particle during the restart?  If so, I agree that this feature will be needed for checkpoint-restart capability.

Reply:
Yes, in that case it would replace the lines just before this line in the script. I think that would make a good runtime option.